src/browse.ts

```diff
@@ -92,6 +92,65 @@ export async function browse (page: Page, website: string, mode: string, tabID:
   return resp.split('\n').filter(line => line.trim() !== '').join('\n')
 }
 
+export async function filterContent (page: Page, tabID: string, printTabID: boolean, filter: string, settings: BrowserSettings): Promise<string> {
+  // Navigate and get the page contents
+  const html = await getPageHTML(page, settings);
+
+  const $ = cheerio.load(html);
+  let filteredContent = '';
+
+  // Check if the filter is an ID, class, or tag selector
+  if (filter.startsWith('#') || filter.startsWith('.') || /^[a-zA-Z]/.test(filter)) {
+    // Use the filter to select elements matching the CSS selector
+    $(filter).each((i, elem) => {
+      filteredContent += $.html(elem);
+    });
+  } else {
+    throw new Error(`Invalid filter format: ${filter}. Use a valid CSS selector.`);
```

Comment:
I think we should make it clear that only ID, class, and tag selectors are supported here:

```suggestion
    throw new Error(`Invalid filter format: ${filter}. Use a CSS ID, class, or tag selector.`);
```